### DISCUSSION TITLE
Use Java 11 as the future baseline?

### DISCUSSION THREAD

**Discussion by phax** [phax/ph-commons#29] [Polls]
Currently ph-commons and all my other ph* libraries use Java 1.8 as the baseline, as it provides all the needed features and has nice coverage. New releases of existing, commonly used libraries, like JAXB, JAX-WS etc. enforce Java 11 as the baseline so the ph* libraries cannot be updated to the latest and greatest version.

Updating to Java 11 as the baseline of all the libraries would mean:
* Java 11 or newer will be required for building the software
* Java 11 or newer will be required for running the software
* the Parent POM will enforce the usage of Java 11
* The latest JAXB, JAX_WS etc libraries can be used as the baseline
* new major releases of all shared libraries as well as all related applications will be created
* Source code backwards compatibility to Java 1.8 will be dropped (this only effects very few places where APIs got Generic parameters  in the runtime)

An update to Java 11 does not mean:
* That all the syntactic sugar of Java 11 (e.g. `var`) will be added to the components

**Comment by rkottmann** [comment]
It is not only ok to move to java 11+, I'd say it would make my life much easier being able to constantly move forward and not being stuck with legacy versions.

**Comment by phax** [comment]
Okay, so the baseline for all ph-projects will shift from Java 8 to Java 11 over the next weeks

**Comment by phax** [comment]
The poll results are:
![grafik](https://github.com/phax/ph-commons/assets/196219/2beac5bf-4c4d-4b7e-81f9-607410d9c687)
